error/s2n_errno.h

```diff
@@ -268,6 +268,8 @@ typedef enum {
     S2N_ERR_INVALID_STATE,
     S2N_ERR_EARLY_DATA_NOT_ALLOWED,
     S2N_ERR_NO_CERT_FOUND,
+    S2N_ERR_CERT_NOT_VALIDATED,
+    S2N_ERR_INVALID_CERT_CHAIN,
```

Comment:
Nit: S2N_ERR_INVALID_ARGUMENT might be sufficient without S2N_ERR_INVALID_CERT_CHAIN, since the problem is just that the customer passed in an argument that didn't meet the API's requirements.